engine/class_modules/priest/sc_priest_pets.cpp

```diff
@@ -1155,6 +1284,51 @@ void priest_t::trigger_shadowflame_prism( player_t* target )
   }
 }
 
+action_t* priest_t::get_living_shadow_action( living_shadow_action action )
+{
+  priest_pet_t* pet = debug_cast<priest_pet_t*>( pets.your_shadow.active_pet() );
+
+  if ( pet )
+  {
+    switch ( action )
+    {
+      case living_shadow_action::SHADOW_SPIKE:
+        return pet->shadow_spike;
+      case living_shadow_action::SHADOW_SPIKE_VOLLEY:
+        return pet->shadow_spike_volley;
+      case living_shadow_action::SHADOW_SEAR:
+        return pet->shadow_sear;
+      case living_shadow_action::SHADOW_NOVA:
+        return pet->shadow_nova;
+    }
+  }
+
+  return nullptr;
+}
+
+void priest_t::trigger_living_shadow_action( player_t* target, living_shadow_action action )
+{
+  auto pet_action = get_living_shadow_action( action );
+
+  if ( pet_action )
+  {
+    pet_action->set_target( target );
+    pet_action->execute();
```

Comment:
The main issue as fasr as I can see is that you call execute() here instead of schedule_execute(). That means you are directly executing the main part of the pet action right here and now (unless there is some travel time), instead of properly scheduling it through the pets action system.

That might definitely explain why you have trouble with channeling actions and the like.

Reply:
@scamille Right now it is working as expected while using execute, as we manually manage cancelling channels by overriding `last_tick` in priest_spell. We need to do this anyways because once the player stops channeling mind flay we need to also cancel the Living Shadows channel. Because of this i dont think we have any issues with `execute` but i could be missing something.

With that being said are there advantages to using the pet action system that we would want to swap this to schedule execute instead?

Reply:
Maybe @navv1234 can better assess the advantages or disadvantages of each approach - I am quite rusty when it comes to player scheduling and all that.